### PR TITLE
Changes to the database (added role entity)

### PR DESCRIPTION
**Pull request by saroskovic**

### CODE REVIEW

File: src/main/java/axiomq/com/starwars/controllers/UserController.java [hunk 1]
```diff
@@ -3,16 +3,21 @@
...
 import java.util.List;
 
 @RestController
 @AllArgsConstructor
+@RequiredArgsConstructor
 @RequestMapping("/user")
 public class UserController {
 
-    private UserService userService;
+    @Autowired
```
Comment (stefanbesovic) [inline src/main/java/axiomq/com/starwars/controllers/UserController.java:18]: Don't use Autowired annotation. Instead use:
private final UserService userService 
with @RequiredArgsConstructor annotation on class.

File: src/main/java/axiomq/com/starwars/entities/User.java [hunk 2]
```diff
@@ -17,28 +18,16 @@ public class User {
...
 
-    @Enumerated(EnumType.STRING)
-    private UserType userType;
+    @ManyToOne
+    @JoinColumn(name = "role_id")
+    private Role role;
+
     private String email;
+
+    @JsonIgnore
```
Comment (stefanbesovic) [inline src/main/java/axiomq/com/starwars/entities/User.java:27]: use dto objects for response and request to freely manipulate with attributes.

File: src/main/java/axiomq/com/starwars/services/implementations/UserServiceImpl.java [hunk 3]
```diff
@@ -49,4 +48,7 @@ public void deleteUser(Long userId) {
         userRepository.delete(user);
 
     }
+
+
```
Comment (stefanbesovic) [inline src/main/java/axiomq/com/starwars/services/implementations/UserServiceImpl.java:52]: try to remove unnecessary spaces. ctrl+alt+L command for refactoring code I think

File: src/main/java/axiomq/com/starwars/entities/User.java [hunk 4]
```diff
@@ -17,28 +18,16 @@ public class User {
...
     private String password;
-    private String username;
 
-    @Override
```
Comment (stefanbesovic) [inline src/main/java/axiomq/com/starwars/entities/User.java:26]: I think you should add equals and hashcode methods.

File: src/main/java/axiomq/com/starwars/StarwarsApplication.java [hunk 5]
```diff
@@ -12,4 +12,5 @@ public static void main(String[] args) {
 
 
 
+
```
Comment (branimir-vujicic) [inline src/main/java/axiomq/com/starwars/StarwarsApplication.java:15]: format files before commit

**Pull request merged**